include/mfmg/dealii/lanczos_deflatedop.templates.hpp

```diff
@@ -0,0 +1,130 @@
+/*************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_LANCZOS_DEFLATEDOP_TEMPLATE_HPP
+#define MFMG_LANCZOS_DEFLATEDOP_TEMPLATE_HPP
+
+#include <cassert>
+#include <cmath>
+
+#include "lanczos_deflatedop.hpp"
+
+namespace mfmg
+{
+
+/// \brief Deflated operator: constructor
+template <typename OperatorType, typename VectorType>
+DeflatedOperator<OperatorType, VectorType>::DeflatedOperator(
+    const OperatorType &base_op)
```

Comment:
Comment that you are using curiously recurring template pattern here.

Reply:
Would you please recall on this thread what are the requirements for `OperatorType`?
`const`-qualified `m()`, `n()`, and `vmult()` member functions.  Is that all?

Reply:
yes.